hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -1967,6 +2035,19 @@ func reconcileHostedControlPlane(hcp *hyperv1.HostedControlPlane, hcluster *hype
 		hcp.Spec.SecretEncryption = hcluster.Spec.SecretEncryption.DeepCopy()
 	}
 
+	// If ROSA passes KubeAPICustomKubeconfig.CustomKubeConfig, we should pass it through to the HCP, otherwise
```

Comment:
what is KubeAPICustomKubeconfig.CustomKubeConfig?

Reply:
This is what I've mentioned above https://github.com/openshift/hypershift/pull/5458#discussion_r1949425238 so maybe I'm wrong. If this is not used by MCE or ROSA, I can remove it.

Reply:
To clarify, I've removed the hcp.spec.KasCustomKubeconfig because it's an old implementation detail that is not relevant anymore and I've changed the PR to simplify the implementation in that way.